### ISSUE TITLE
building image fails on Mac if run in parallel

### ISSUE THREAD

**Issue by gergelyfabian** [bazel-contrib/rules_oci#462]
I'm running a `bazel build //...` on a monorepo on a Mac, and it fails with:

```
bazel-out/darwin_arm64-fastbuild/bin/my_module/image_image.sh: line 160: external/jq_darwin_arm64/jq: Bad CPU type in executable
```

If I execute a single target (the same that fails when building all targets), then it passes: `bazel build //my_module:image`

Details:

```
ERROR: /Users/user/my_monorepo/my_module/BUILD:199:10: OCI Image //my_module:image failed: (Exit 126): image_image.sh failed: error executing OCIImage command (from target //my_module:image) bazel-out/darwin_arm64-fastbuild/bin/my_module/image_image.sh mutate oci:layout/bazel-out/darwin_arm64-fastbuild/bin/external/my_base_single/layout ... (remaining 10 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
2024/01/16 15:21:29 serving on port 51805
2024/01/16 15:21:29 GET /v2/
2024/01/16 15:21:29 HEAD /v2/oci/layout/manifests/latest 404 NAME_UNKNOWN Unknown name
2024/01/16 15:21:29 HEAD /v2/oci/layout/blobs/sha256:c0961fc0219e94266400e65086cc2b8bf6b5599c0fcf32b5a234a289bff744cf 404 BLOB_UNKNOWN Unknown blob
2024/01/16 15:21:29 HEAD /v2/oci/layout/blobs/sha256:44fdcdbf0add8a6bb35e8b02659baa2a99d6f02b98fe447eaa108003cfe6c486 404 BLOB_UNKNOWN Unknown blob
2024/01/16 15:21:29 HEAD /v2/oci/layout/blobs/sha256:f1a4db3e1dd59b465e3e42daa3cac72917640bc0b3567a592c13c302c7cdbcae 404 BLOB_UNKNOWN Unknown blob
2024/01/16 15:21:29 HEAD /v2/oci/layout/blobs/sha256:cff675280b647e97abb3c1cf7ecb09abc4492a73e849347d8e8d2ec1e7d9b927 404 BLOB_UNKNOWN Unknown blob
2024/01/16 15:21:29 POST /v2/oci/layout/blobs/uploads/
2024/01/16 15:21:29 POST /v2/oci/layout/blobs/uploads/
2024/01/16 15:21:29 POST /v2/oci/layout/blobs/uploads/
2024/01/16 15:21:29 POST /v2/oci/layout/blobs/uploads/
2024/01/16 15:21:29 PATCH /v2/oci/layout/blobs/uploads/8879807339482372859
2024/01/16 15:21:29 PATCH /v2/oci/layout/blobs/uploads/266746645724216961
2024/01/16 15:21:29 PATCH /v2/oci/layout/blobs/uploads/6934939217271212404
2024/01/16 15:21:30 PATCH /v2/oci/layout/blobs/uploads/3160380437138956462
2024/01/16 15:21:30 PUT /v2/oci/layout/blobs/uploads/8879807339482372859?digest=sha256%3Ac0961fc0219e94266400e65086cc2b8bf6b5599c0fcf32b5a234a289bff744cf
2024/01/16 15:21:30 PUT /v2/oci/layout/blobs/uploads/266746645724216961?digest=sha256%3A44fdcdbf0add8a6bb35e8b02659baa2a99d6f02b98fe447eaa108003cfe6c486
2024/01/16 15:21:30 HEAD /v2/oci/layout/blobs/sha256:e4b8d6a4f0bf88599dfb4d6f1d4fb642f5a36bd4ea2139789d111cb4d302290a 404 BLOB_UNKNOWN Unknown blob
2024/01/16 15:21:30 POST /v2/oci/layout/blobs/uploads/
2024/01/16 15:21:30 HEAD /v2/oci/layout/blobs/sha256:70e1953fadca563137f6931a0ad795758ea6ec5ca159203a8c292102633cb364 404 BLOB_UNKNOWN Unknown blob
2024/01/16 15:21:30 PUT /v2/oci/layout/blobs/uploads/6934939217271212404?digest=sha256%3Af1a4db3e1dd59b465e3e42daa3cac72917640bc0b3567a592c13c302c7cdbcae
2024/01/16 15:21:30 POST /v2/oci/layout/blobs/uploads/
2024/01/16 15:21:30 PUT /v2/oci/layout/blobs/uploads/3160380437138956462?digest=sha256%3Acff675280b647e97abb3c1cf7ecb09abc4492a73e849347d8e8d2ec1e7d9b927
2024/01/16 15:21:30 HEAD /v2/oci/layout/blobs/sha256:4f56adab6a3253ba68cfdbe3a1bd10d7e789a879d3e716075ff0ffe5833fd82d 404 BLOB_UNKNOWN Unknown blob
2024/01/16 15:21:30 PATCH /v2/oci/layout/blobs/uploads/1292643695001537923
2024/01/16 15:21:30 PATCH /v2/oci/layout/blobs/uploads/3514861300539231974
2024/01/16 15:21:30 POST /v2/oci/layout/blobs/uploads/
2024/01/16 15:21:30 PUT /v2/oci/layout/blobs/uploads/3514861300539231974?digest=sha256%3A70e1953fadca563137f6931a0ad795758ea6ec5ca159203a8c292102633cb364
2024/01/16 15:21:30 PUT /v2/oci/layout/blobs/uploads/1292643695001537923?digest=sha256%3Ae4b8d6a4f0bf88599dfb4d6f1d4fb642f5a36bd4ea2139789d111cb4d302290a
2024/01/16 15:21:30 PATCH /v2/oci/layout/blobs/uploads/5720809939016621372
2024/01/16 15:21:30 PUT /v2/oci/layout/blobs/uploads/5720809939016621372?digest=sha256%3A4f56adab6a3253ba68cfdbe3a1bd10d7e789a879d3e716075ff0ffe5833fd82d
2024/01/16 15:21:30 PUT /v2/oci/layout/manifests/latest
/var/folders/01/x9vtqz0s2ms5ft4yyqxy6kgm0000gn/T/tmp.jYUMx0T0
2024/01/16 15:21:30 GET /v2/
2024/01/16 15:21:30 GET /v2/oci/layout/manifests/sha256:c102dd158db917e6d81e23517fb5b60af25e8815313898ba3af5b1a40a8128b2
2024/01/16 15:21:30 GET /v2/oci/layout/blobs/sha256:4f56adab6a3253ba68cfdbe3a1bd10d7e789a879d3e716075ff0ffe5833fd82d
2024/01/16 15:21:30 GET /v2/
2024/01/16 15:21:30 HEAD /v2/oci/layout/manifests/sha256:eec1cb0f5116b3786d8196462a9e5494da856f2bcef1c05bbd3345eb211d0cd0 404 MANIFEST_UNKNOWN Unknown manifest
2024/01/16 15:21:30 HEAD /v2/oci/layout/blobs/sha256:c0961fc0219e94266400e65086cc2b8bf6b5599c0fcf32b5a234a289bff744cf
2024/01/16 15:21:30 existing blob: sha256:c0961fc0219e94266400e65086cc2b8bf6b5599c0fcf32b5a234a289bff744cf
2024/01/16 15:21:30 HEAD /v2/oci/layout/blobs/sha256:cff675280b647e97abb3c1cf7ecb09abc4492a73e849347d8e8d2ec1e7d9b927
2024/01/16 15:21:30 HEAD /v2/oci/layout/blobs/sha256:f1a4db3e1dd59b465e3e42daa3cac72917640bc0b3567a592c13c302c7cdbcae
2024/01/16 15:21:30 HEAD /v2/oci/layout/blobs/sha256:44fdcdbf0add8a6bb35e8b02659baa2a99d6f02b98fe447eaa108003cfe6c486
2024/01/16 15:21:30 HEAD /v2/oci/layout/blobs/sha256:e4b8d6a4f0bf88599dfb4d6f1d4fb642f5a36bd4ea2139789d111cb4d302290a
2024/01/16 15:21:30 existing blob: sha256:e4b8d6a4f0bf88599dfb4d6f1d4fb642f5a36bd4ea2139789d111cb4d302290a
2024/01/16 15:21:30 existing blob: sha256:cff675280b647e97abb3c1cf7ecb09abc4492a73e849347d8e8d2ec1e7d9b927
2024/01/16 15:21:30 existing blob: sha256:f1a4db3e1dd59b465e3e42daa3cac72917640bc0b3567a592c13c302c7cdbcae
2024/01/16 15:21:30 existing blob: sha256:44fdcdbf0add8a6bb35e8b02659baa2a99d6f02b98fe447eaa108003cfe6c486
2024/01/16 15:21:30 HEAD /v2/oci/layout/blobs/sha256:70e1953fadca563137f6931a0ad795758ea6ec5ca159203a8c292102633cb364
2024/01/16 15:21:30 HEAD /v2/oci/layout/blobs/sha256:98b489bc2b84c55b02bafe7e5ab09f82bbbac214a91d1981105e3db31ae7ec5e 404 BLOB_UNKNOWN Unknown blob
2024/01/16 15:21:30 HEAD /v2/oci/layout/blobs/sha256:107484f2fcc2519d8c85360dd38c4d71f18890d721bffcb01a367b8da1acd9a4 404 BLOB_UNKNOWN Unknown blob
2024/01/16 15:21:30 HEAD /v2/oci/layout/blobs/sha256:69fdcc76593bf398ad5a8f17ef40a192cf404fb58b192b2059e7caac547417a2 404 BLOB_UNKNOWN Unknown blob
2024/01/16 15:21:30 existing blob: sha256:70e1953fadca563137f6931a0ad795758ea6ec5ca159203a8c292102633cb364
2024/01/16 15:21:30 POST /v2/oci/layout/blobs/uploads/
2024/01/16 15:21:30 POST /v2/oci/layout/blobs/uploads/
2024/01/16 15:21:30 POST /v2/oci/layout/blobs/uploads/
2024/01/16 15:21:30 HEAD /v2/oci/layout/blobs/sha256:575951456a45051e258aaf0b6c03d60de1c7c5ca27d3a3d7093647ca4ee91c58 404 BLOB_UNKNOWN Unknown blob
2024/01/16 15:21:30 POST /v2/oci/layout/blobs/uploads/
2024/01/16 15:21:30 PATCH /v2/oci/layout/blobs/uploads/7990440352018795697
2024/01/16 15:21:30 PATCH /v2/oci/layout/blobs/uploads/5763081549225214992
2024/01/16 15:21:30 PATCH /v2/oci/layout/blobs/uploads/3725641611851077439
2024/01/16 15:21:30 PATCH /v2/oci/layout/blobs/uploads/4649294044216256920
2024/01/16 15:21:30 PUT /v2/oci/layout/blobs/uploads/7990440352018795697?digest=sha256%3A107484f2fcc2519d8c85360dd38c4d71f18890d721bffcb01a367b8da1acd9a4
2024/01/16 15:21:30 PUT /v2/oci/layout/blobs/uploads/5763081549225214992?digest=sha256%3A98b489bc2b84c55b02bafe7e5ab09f82bbbac214a91d1981105e3db31ae7ec5e
2024/01/16 15:21:30 pushed blob: sha256:98b489bc2b84c55b02bafe7e5ab09f82bbbac214a91d1981105e3db31ae7ec5e
2024/01/16 15:21:30 pushed blob: sha256:107484f2fcc2519d8c85360dd38c4d71f18890d721bffcb01a367b8da1acd9a4
2024/01/16 15:21:30 PUT /v2/oci/layout/blobs/uploads/3725641611851077439?digest=sha256%3A69fdcc76593bf398ad5a8f17ef40a192cf404fb58b192b2059e7caac547417a2
2024/01/16 15:21:30 HEAD /v2/oci/layout/blobs/sha256:041091c3b6fc6535154d81bdd59c3358fcd2b9146740fd036705bb4b5dfc3759 404 BLOB_UNKNOWN Unknown blob
2024/01/16 15:21:30 PUT /v2/oci/layout/blobs/uploads/4649294044216256920?digest=sha256%3A575951456a45051e258aaf0b6c03d60de1c7c5ca27d3a3d7093647ca4ee91c58
2024/01/16 15:21:30 pushed blob: sha256:69fdcc76593bf398ad5a8f17ef40a192cf404fb58b192b2059e7caac547417a2
2024/01/16 15:21:30 POST /v2/oci/layout/blobs/uploads/
2024/01/16 15:21:30 pushed blob: sha256:575951456a45051e258aaf0b6c03d60de1c7c5ca27d3a3d7093647ca4ee91c58
2024/01/16 15:21:30 PATCH /v2/oci/layout/blobs/uploads/547009175369324414
2024/01/16 15:21:30 PUT /v2/oci/layout/blobs/uploads/547009175369324414?digest=sha256%3A041091c3b6fc6535154d81bdd59c3358fcd2b9146740fd036705bb4b5dfc3759
2024/01/16 15:21:30 pushed blob: sha256:041091c3b6fc6535154d81bdd59c3358fcd2b9146740fd036705bb4b5dfc3759
2024/01/16 15:21:30 PUT /v2/oci/layout/manifests/sha256:eec1cb0f5116b3786d8196462a9e5494da856f2bcef1c05bbd3345eb211d0cd0
2024/01/16 15:21:30 127.0.0.1:51805/oci/layout@sha256:eec1cb0f5116b3786d8196462a9e5494da856f2bcef1c05bbd3345eb211d0cd0: digest: sha256:eec1cb0f5116b3786d8196462a9e5494da856f2bcef1c05bbd3345eb211d0cd0 size: 1894
bazel-out/darwin_arm64-fastbuild/bin/my_module/image_image.sh: line 160: external/jq_darwin_arm64/jq: Bad CPU type in executable
2024/01/16 15:21:30 GET /v2/
2024/01/16 15:21:30 GET /v2/
2024/01/16 15:21:30 GET /v2/oci/layout/manifests/sha256:eec1cb0f5116b3786d8196462a9e5494da856f2bcef1c05bbd3345eb211d0cd0
2024/01/16 15:21:30 GET /v2/oci/layout/manifests/sha256:eec1cb0f5116b3786d8196462a9e5494da856f2bcef1c05bbd3345eb211d0cd0
2024/01/16 15:21:30 GET /v2/oci/layout/blobs/sha256:041091c3b6fc6535154d81bdd59c3358fcd2b9146740fd036705bb4b5dfc3759
2024/01/16 15:21:30 GET /v2/
2024/01/16 15:21:30 HEAD /v2/oci/layout/blobs/sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855 404 BLOB_UNKNOWN Unknown blob
2024/01/16 15:21:30 POST /v2/oci/layout/blobs/uploads/
2024/01/16 15:21:30 PATCH /v2/oci/layout/blobs/uploads/4735489431635866533
2024/01/16 15:21:30 PUT /v2/oci/layout/blobs/uploads/4735489431635866533?digest=sha256%3Ae3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
2024/01/16 15:21:30 pushed blob: sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
2024/01/16 15:21:30 HEAD /v2/oci/layout/manifests/sha256:dc36c0cd2f03312029c3a8fb2abe23eef00d7cad92fafcea767f0b2d4515a1db 404 MANIFEST_UNKNOWN Unknown manifest
2024/01/16 15:21:30 PUT /v2/oci/layout/manifests/sha256:dc36c0cd2f03312029c3a8fb2abe23eef00d7cad92fafcea767f0b2d4515a1db
2024/01/16 15:21:30 127.0.0.1:51805/oci/layout@sha256:dc36c0cd2f03312029c3a8fb2abe23eef00d7cad92fafcea767f0b2d4515a1db: digest: sha256:dc36c0cd2f03312029c3a8fb2abe23eef00d7cad92fafcea767f0b2d4515a1db size: 1891
```

**Comment by gergelyfabian** [comment]
If I install Rosetta, then it works:

```
 /usr/sbin/softwareupdate --install-rosetta --agree-to-license
```

I have a Mac with Apple M1 silicon.

**Comment by thesayyn** [comment]
yeah, we are aware of this issue. jq which we depend on doesn't have a darwin/arm64 that forces users to install rosetta. This is already fixed in bazel-lib 2.0 but we haven't upgraded  yet.

**Comment by thesayyn** [comment]
fixed by https://github.com/bazel-contrib/rules_oci/pull/550
